rollup/fees/rollup_fee.go

```diff
@@ -210,12 +214,21 @@ func calculateEncodedL1DataFeeFeynman(
 	execScalar *big.Int,
 	blobScalar *big.Int,
 ) *big.Int {
-	// tx size (RLP-encoded)
 	txSize := big.NewInt(int64(len(data)))
 
-	// compression_ratio(tx) = 1 (placeholder, scaled to match scalars precision)
+	// Default compression ratio is 1.0 (no compression)
 	compressionRatio := big.NewInt(rcfg.Precision.Int64())
 
+	compressedBytes, err := zstd.CompressScrollBatchBytes(data)
+	if err != nil {
+		log.Error("Batch compression failed, using 1.0 compression ratio", "err", err)
```

Comment:
adjust the log message and would be good to also add the tx hash to the log message here to make it more meaningful.